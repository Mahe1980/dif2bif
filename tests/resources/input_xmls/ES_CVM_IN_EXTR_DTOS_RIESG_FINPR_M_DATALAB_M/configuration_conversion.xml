<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<configuration verbose="false" debugMode="false">
    <generalConfiguration>
        <properties>
            <property name="hdfsRootPath" value="/data/raw"/>
            <property name="localRootPath" value="/var/SP/data/staging"/>
        </properties>
    </generalConfiguration>
    <dataConversions>
        <dataConversion configurationId="cba8204c-0385-4c82-a748-f85b6f7adb56" description="ES_CVM_IN_EXTR_DTOS_RIESG_FINPR_M_DATALAB_M Conversion" name="ES_CVM_IN_EXTR_DTOS_RIESG_FINPR_M_DATALAB_M">
            <properties>
                <property name="country" value="es"></property>
                <property name="datasource" value="cvm"></property>
                <property name="interface" value="ES_CVM_IN_EXTR_DTOS_RIESG_FINPR_M_DATALAB_M"></property>
                <property name="version" value="1.1"></property>
                <property name="contains_header" value="false"></property>
                <property name="DeleteInputDataAfterSuccessfulConversion" value="false"></property>
            </properties>
            <inputFileSelectionCriteria>
                <criteria name="MinimumNumberOfFiles" value="1"></criteria>
            </inputFileSelectionCriteria>
            <dataRoot path="${hdfsRootPath}/vf_${country}/${datasource}/${interface}/${version}"></dataRoot>
            <conversionActions>
                <readAction name="ReadFromCsvAction">
                    <files type="delimited" separator="\t"/>
                </readAction>
                <conversionAction name="BasicConversionAction"></conversionAction>
                <writeAction name="WriteToParquetAction">
                    <hive>
                        <database name="raw_es"></database>
                        <tableName name="cvm_ES_CVM_IN_EXTR_DTOS_RIESG_FINPR_M_DATALAB_M"></tableName>
                    </hive>
                </writeAction>
            </conversionActions>
            <schema>
                <field name="MSISDN" type="string" required="false"/>
                <field name="NIF" type="string" required="false"/>
                <field name="NUMERO_CUENTA" type="string" required="false"/>
                <field name="ESTADO_SERVICIO" type="string" required="false"/>
                <field name="CODIGO_DESCUENTO" type="string" required="false"/>
                <field name="DESCRIPCION_DESCUENTO" type="string" required="false"/>
                <field name="ESTADO_DESCUENTO" type="string" required="false"/>
                <field name="FX_ACTIVACION_DTO" type="timestamp" required="false" format="yyyyMMdd"/>
                <field name="FX_DESACTIVACION_DTO" type="timestamp" required="false" format="yyyyMMdd"/>
            </schema>
            <partitions>
                <partition name="year" fieldType="integer" position="0:0" source="filepath:year"></partition>
                <partition name="month" fieldType="integer" position="0:0" source="filepath:month"></partition>
                <partition name="day" fieldType="integer" position="last" source="datetime"></partition>
            </partitions>
        </dataConversion>
	</dataConversions>
</configuration>

