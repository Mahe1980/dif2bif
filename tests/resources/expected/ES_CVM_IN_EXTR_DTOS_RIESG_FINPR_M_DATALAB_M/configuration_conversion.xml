<configuration verbose="false" debugMode="false">
  <generalConfiguration>
    <properties>
      <property name="rootPath" value="gs://vfes-dh-rawingested"/>
      <property name="gcsOutputRootPath" value="gs://vfes-dh-cvm-rawprepared"/>
    </properties>
  </generalConfiguration>
  <dataConversions>
    <dataConversion configurationId="cba8204c-0385-4c82-a748-f85b6f7adb56" description="ES_CVM_IN_EXTR_DTOS_RIESG_FINPR_M_DATALAB_M Conversion" name="ES_CVM_IN_EXTR_DTOS_RIESG_FINPR_M_DATALAB_M">
      <properties>
        <property name="country" value="es"/>
        <property name="datasource" value="cvm"/>
        <property name="interface" value="ES_CVM_IN_EXTR_DTOS_RIESG_FINPR_M_DATALAB_M"/>
        <property name="version" value="1.1"/>
        <property name="DeleteInputDataAfterSuccessfulConversion" value="false"/>
        <property name="bigQueryProject" value="dh"/>
        <property name="bigQueryDataCategory" value="cvm"/>
      </properties>
      <inputFileSelectionCriteria>
        <criteria name="MinimumNumberOfFiles" value="1"/>
      </inputFileSelectionCriteria>
      <dataRoot path="${rootPath}/${datasource}/${interface}/${version}"/>
      <conversionActions>
        <readAction name="ReadFromCsvAction">
          <files type="delimited" separator="\t" contains_header="false" charset="utf-8"/>
        </readAction>
        <conversionAction name="BasicConversionAction"/>
        <writeActions>
          <writeAction name="WriteToBigQueryAction">
            <config>
              <dataset name="vf${country}_${bigQueryProject}_lake_${bigQueryDataCategory}_rawprepared_s"/>
              <tableName name="es_cvm_in_extr_dtos_riesg_finpr_m_datalab_m"/>
            </config>
          </writeAction>
          <writeAction name="WriteToParquetAction">
            <config>
              <path>${gcsOutputRootPath}/${interface}/${version}/parquet</path>
            </config>
          </writeAction>
        </writeActions>
      </conversionActions>
      <schema>
        <field name="MSISDN" type="string" required="false"/>
        <field name="NIF" type="string" required="false"/>
        <field name="NUMERO_CUENTA" type="string" required="false"/>
        <field name="ESTADO_SERVICIO" type="string" required="false"/>
        <field name="CODIGO_DESCUENTO" type="string" required="false"/>
        <field name="DESCRIPCION_DESCUENTO" type="string" required="false"/>
        <field name="ESTADO_DESCUENTO" type="string" required="false"/>
        <field name="FX_ACTIVACION_DTO" type="timestamp" required="false" format="yyyyMMdd"/>
        <field name="FX_DESACTIVACION_DTO" type="timestamp" required="false" format="yyyyMMdd"/>
      </schema>
      <partitions>
        <partition name="fileRegex" source="property" position="ES.CVM.(PREPFINAL|EXTRNIFS|POSTFINALACEXT|PREPTAR|IN_EXTR_AC_FINAL_POSPAGO_DIR_CORREGIDA|IN_EXTR_DTOS_RIESG_FINPR_F|IN_EXTR_DTOS_RIESG_FINPR_M).\w+.M.(\d{2})(\d{2})(\d{4})\d{0,}.gz"/>
        <partition name="4" source="property" position="year"/>
        <partition name="3" source="property" position="month"/>
        <partition name="day" source="datetime_now"/>
      </partitions>
    </dataConversion>
  </dataConversions>
</configuration>
